src/imex/transfer.rs

```diff
@@ -475,14 +475,15 @@ async fn on_blob(
         let token = ticket.token().to_string();
         jobs.lock().await.spawn(async move {
             if let Err(err) = context.sql.import(&path, token).await {
-                error!(context, "cannot import database: {:#?}", err);
+                error!(context, "cannot import database: {err:#}");
+            } else if let Err(err) = imex::adjust_delete_server_after(&context).await {
```

Comment:
Yes, this is only for old backups (or for backup transfers from older versions). This isn't reliable because if we crash before "adjusting" the setting, after a restart this isn't retried. But this is unlikely enough and only a transient problem (until users are updated)